src/lava/proc/graded/models.py

```diff
@@ -0,0 +1,148 @@
+# Copyright (C) 2023 Intel Corporation
+# SPDX-License-Identifier: BSD-3-Clause
+# See: https://spdx.org/licenses/
+
+import numpy as np
+import typing as ty
+
+from lava.magma.core.sync.protocols.loihi_protocol import LoihiProtocol
+from lava.magma.core.model.py.ports import PyInPort, PyOutPort
+from lava.magma.core.model.py.type import LavaPyType
+from lava.magma.core.resources import CPU
+from lava.magma.core.decorator import implements, requires, tag
+from lava.magma.core.model.py.model import PyLoihiProcessModel
+
+from lava.proc.graded.process import GradedVec, NormVecDelay, InvSqrt
+
+
+class AbstractGradedVecModel(PyLoihiProcessModel):
+    a_in = None
+    s_out = None
+
+    v = None
+    vth = None
+    exp = None
+
+    def run_spk(self) -> None:
+        a_in_data = self.a_in.recv()
+        self.v += a_in_data
+
+        is_spike = np.abs(self.v) > self.vth
+        sp_out = self.v * is_spike
+
+        self.v[:] = 0
+
+        self.s_out.send(sp_out)
+
+
+@implements(proc=GradedVec, protocol=LoihiProtocol)
+@requires(CPU)
+@tag('fixed_pt')
+class PyGradedVecModelFixed(AbstractGradedVecModel):
+    """Fixed point implementation of GradedVec"""
```

Comment:
Ideally, this would explain what this does, not just what it implements.

Reply:
i mean half the docstrings for the models are just like this i just followed their formatting